controllers/update.go

```diff
@@ -499,7 +503,11 @@ func (r *ComponentDetectionQueryReconciler) updateComponentStub(req ctrl.Request
 			Revision:      componentDetectionQuery.Spec.GitSource.Revision,
 			DockerfileURL: link,
 		}
-		componentName := r.getComponentName(log, ctx, req.Namespace, gitSource)
+		generateComponentName := false
+		if componentDetectionQuery.Spec.GenerateComponentName {
+			generateComponentName = componentDetectionQuery.Spec.GenerateComponentName
+		}
```

Comment:
cant this block be declared once for both devfile map and dockerfile map?

Reply:
they are in different functions. updated the PR to move this block to `getComponentName` method.